src/nvim/lua/executor.c

```diff
@@ -1809,3 +1810,64 @@ void nlua_execute_log_keystroke(int c)
 #endif
 }
 
+// Checks if str is in blacklist array
+bool is_in_blacklist(const char *str, char *blacklist[], int blacklist_size)
+{
+  for (int i = 0; i < blacklist_size; i++) {
+    if (strcmp(blacklist[i], str) == 0) {
+      return true;
+    }
+  }
+  return false;
+}
+// Get sctx of current file being sourc3d if doesn't exsist genarate it
+static sctx_T *nlua_get_sourcing_sctx(void)
+{
+  lua_State *const lstate = global_lstate;
+  sctx_T *retval = (sctx_T *)xmalloc(sizeof(sctx_T));
+  retval->sc_seq = -1;
+  retval->sc_sid = SID_LUA;
+  retval->sc_lnum = -1;
+  lua_Debug *info = (lua_Debug *)xmalloc(sizeof(lua_Debug));
+
+  // Files where internal wrappers are defined so we can ignore them
+  // like vim.o/opt etc are defined in _meta.lua
+  char *blacklist[] = {
+    "vim/_meta.lua",
```

Comment:
@bfredl @justinmk Should we let users a way to append to this list?

For example if user is using a wrapper around api / a plugin all the verbose result
will point to that wrapper/plugin location .

I'm not sure adding an option for this is worth it or not since it'll be only for very specific case in lua.

Reply:
Not really sure we want this at all. We're already iterating the stacktrace... why are we throwing away that gold? We could show the full stacktrace after the `Last set from...` line.

We can truncate long stacktraces. If stacktraces tend to be deep, then we can start filtering out useless parts.

Reply:
We can show complete traceback . Initially I didn't because the more I unwinded the stack the slower the startup was . Since it's now optional and only enabled if user choses so startuptime doesn't matter .

lua has a `debug.traceback` function I wonder if it's exposed to c .

Reply:
we don't necessarily need to go in that direction. I'm just saying, there is no need for a "blacklist" concept. We are already iterating `lua_getstack`, right? Just collect the results into a string.

Reply:
One thing though we don't set last set message here . We just add line no & filename info in sctx . The last set message is constructed when verbose is used .

Should I add a traceback field to sctx ?

Reply:
@justinmk  Current behavior is probably fine as 
it's same as viml.
```vim
function! Map(lhs, rhs) abort
  exe 'nnoremap '.a:lhs.' '.a:rhs
endf

call Map('<leader>aa', ':echo "hello"<cr>')
```
Running `:verb nmap <leader>aa`
shows it's defined is line 2 .

Reply:
> Should I add a traceback field to sctx ?
> 
> script context doesn't sound like a place where you'd put tracebacks

Sure, with verbose=1 why not? Can be truncated to a max size. But that can be a followup PR.

> @justinmk Current behavior is probably fine as viml behaves the same.

Eh, the vim behavior is just whatever was implemented decades ago. It can be improved.